tests/e2e/common/diagnosticcollector.go

```diff
@@ -0,0 +1,564 @@
+			err = fmt.Errorf("%v+ error writing to stderr log file: %v", err, lerr)
+		}
+		return err
+	}
+
+	podsFile, err := os.Create(filepath.Join(d.artifactDir, "pods.yaml"))
+	if err != nil {
+		if _, lerr := d.errLogFile.WriteString(fmt.Sprintf("Error creating pods.yaml: %v\n", err)); lerr != nil {
+			err = fmt.Errorf("%v+ error writing to stderr log file: %v", err, lerr)
+		}
+		return err
+	}
+	defer podsFile.Close()
+
+	data, err := yaml.Marshal(pods)
+	if err != nil {
+		if _, lerr := d.errLogFile.WriteString(fmt.Sprintf("Error marshalling pods: %v\n", err)); lerr != nil {
+			err = fmt.Errorf("%v+ error writing to stderr log file: %v", err, lerr)
+		}
+		return err
+	}
+
+	_, err = podsFile.Write(data)
+	if err != nil {
+		if _, lerr := d.errLogFile.WriteString(fmt.Sprintf("Error writing pods.yaml: %v\n", err)); lerr != nil {
+			err = fmt.Errorf("%v+ error writing to stderr log file: %v", err, lerr)
+		}
+		return err
+	}
+
+	// Get all logs per pod
+	for _, pod := range pods.Items {
+		podLogFile, err := os.Create(filepath.Join(d.artifactDir, fmt.Sprintf("%s_logs.log", pod.Name)))
+		if err != nil {
+			if _, lerr := d.errLogFile.WriteString(fmt.Sprintf("Error creating %s_logs.log: %v\n", pod.Name, err)); lerr != nil {
+				err = fmt.Errorf("%v+ error writing to stderr log file: %v", err, lerr)
+			}
+			return err
+		}
+		defer podLogFile.Close()
+
+		podLogOpts := v1.PodLogOptions{}
+		req := d.clientset.CoreV1().Pods(d.namespace).GetLogs(pod.Name, &podLogOpts)
+		podLogs, err := req.Stream(ctx)
+		if err != nil {
+			if _, lerr := d.errLogFile.WriteString(fmt.Sprintf("Error getting pod logs: %v\n", err)); lerr != nil {
+				err = fmt.Errorf("%v+ error writing to stderr log file: %v", err, lerr)
+			}
+			return err
+		}
+		defer podLogs.Close()
+
+		buf := make([]byte, 4096)
+		for {
+			n, err := podLogs.Read(buf)
+			if err != nil {
+				break
+			}
+			_, err = podLogFile.Write(buf[:n])
+			if err != nil {
+				if _, lerr := d.errLogFile.WriteString(fmt.Sprintf("Error writing pod logs: %v\n", err)); lerr != nil {
+					err = fmt.Errorf("%v+ error writing to stderr log file: %v", err, lerr)
+				}
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// Jobs is a collectorFunc that collects job objects
+func (d *DiagnosticCollector) Jobs(ctx context.Context) error {
+	jobs, err := d.clientset.BatchV1().Jobs(d.namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		if _, lerr := d.errLogFile.WriteString(fmt.Sprintf("Error getting jobs: %v\n", err)); lerr != nil {
+			err = fmt.Errorf("%v+ error writing to stderr log file: %v", err, lerr)
+		}
+		return err
+	}
+
+	jobsFile, err := os.Create(filepath.Join(d.artifactDir, "jobs.yaml"))
+	if err != nil {
+		if _, lerr := d.errLogFile.WriteString(fmt.Sprintf("Error creating jobs.yaml: %v\n", err)); lerr != nil {
+			err = fmt.Errorf("%v+ error writing to stderr log file: %v", err, lerr)
+		}
+		return err
+	}
+	defer jobsFile.Close()
+
+	data, err := yaml.Marshal(jobs)
```

Comment:
Do jobs also have logs like pods?

Reply:
jobs create pods, so the function Pods will collect the logs. no need to do a log collection here

Reply:
Out of scope for this PR: Do we need to store the relationship between the job and the pods that it may create?

Reply:
ummm it would require some work, but it could be helpful once we move into more complex text matrix with multiple tests and jobs. totally a followup